docs/decisions/0006-pluggable-lti-configuration.rst

```diff
@@ -0,0 +1,121 @@
+
+#. Each LTI tool needs to be configured individually, sometimes requiring multiple valiables to be set to get
+a working integration. This makes the course creation process cumbersome on courses that heavily rely on LTI 
+components. 
+#. There's no support for instance-wide LTI tool configuration or any sort of pluggable LTI configurations.
+
+
+This proposal aims to solve these issues by introducing a mechanism to allow LTI configuration pluggability 
+and changes to the XBlock to allow re-using configurations to reduce instructor friction.
+
+
+Decision
+========
+
+The decision is to enable LTI reusability and pluggability using a filter hook (as defined in `OEP-50`_) to fetch re-usable LTI configurations
+from external plugins. 
+
+With this implementation the plugins are responsible for storing the LTI configuration and credentials, while the LTI consumer handles displaying the content.
+
+To enable this, the LTI Configuration model needs to be modified to:
+
+#. Allow externally configured tools (a new :code:`CONFIG_EXTERNAL` choice for the configuration storage).
+#. Two new fields to store an unique external tool id and the plugin name.
+#. Changes to the :code:`get_lti_consumer` methods to allow instancing a :code:`LtiConsumer` from externally managed configurations.
+
+The filter will allow the XBlock LTI Consumer to query re-usable and/or plugabble LTI configurations and display them to course 
+creators as configuration options in Studio. 
+The filter response should contain all the necessary information for the XBlock to locate and and use the externally managed LTI configuration.
+
+The inputs for the filter pipeline should be one of the following:
+
+* The course ID, so that each filter is able to evaluate and return only configurations available to that context
+* The unique configuration ID, so the filter returns exactly the configuration required.
+
+
+Since filter responses are additive (TODO: reference), all filters will create keys with their names in the return dictionary.
+
+.. code-block:: json
+    {
+        "some_plugin": { // Filter should return a dictionary
+            "available_configurations": [
+                {
+                    "lti_version": "LTI_1P1",
+                    "display_name": "Reusable LTI tool 1",
+                    "uuid": some-unique-uuid,
+                    "consumer_params": {
+                        // LTI consumer parameters and configuration
+                    }
+                }
+                {
+                    "lti_version": "LTI_1P3",
+                    "display_name": "Reusable LTI tool 2 - LTI 1.3",
+                    "uuid": another-unique-uuid,
+                    "consumer_params": {
+                        // LTI consumer parameters and configuration
+                    }     
+                }
+            ]
+        },
+        "another_plugin": {
+            "available_configurations": [...]
+        },
+        ...
+    }
```

Comment:
@xitij2000 Makes sense to store the configuration on a single field. :+1:
The unique is basically a UUID that is managed by the external tool/plugin, so to avoid the remote possibility of a conflict between two plugins, I've added the extra field.
I'll change the format to `plugin_name:uuid` (plugin_name as in the pipeline function name that is used).